exporter/context.go

```diff
@@ -374,14 +374,30 @@ func (ic *importContext) Run() error {
 	ic.startImportChannels()
 
 	// Start listing of objects
+	listWorkspaceObjectsAlreadyRunning := false
 	for rnLoop, irLoop := range ic.Importables {
 		resourceName := rnLoop
 		ir := irLoop
+		// TODO: extend this to other services?  Like, Git Folders
+		if !ic.accountLevel && (ir.Service == "notebooks" || ir.Service == "wsfiles" || (ir.Service == "directories" && !ic.incremental)) {
+			if _, exists := ic.listing[ir.Service]; exists && !listWorkspaceObjectsAlreadyRunning {
+				ic.waitGroup.Add(1)
+				log.Printf("[DEBUG] Starting listing of workspace objects")
+				go func() {
+					if err := listWorkspaceObjects(ic); err != nil {
+						log.Printf("[ERROR] listing of workspace objects failed %s", err)
+					}
+					log.Print("[DEBUG] Finished listing of workspace objects")
+					ic.waitGroup.Done()
+				}()
+				listWorkspaceObjectsAlreadyRunning = true
```

Comment:
In case of err on line 388, this would still be set to true right? which shouldn't be the case.